File: backend/src/main/java/com/xuanjian/entity/Technique.java
package com.xuanjian.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("technique")
public class Technique {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long projectId;
    private String name;
    private String grade;
    private String category;
    private String relatedFoundation;
    private String relatedAbility;
    private String knownUsers;
    private String description;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/api/technique.ts
import api from './index';

export interface Technique {
  id: number; name: string; grade: string | null; category: string | null;
  relatedFoundation: string | null; relatedAbility: string | null;
  knownUsers: string | null; description: string | null;
}
export interface Artifact {
  id: number; name: string; type: string | null; creator: string | null;
  effect: string | null; relatedCharacters: string | null; description: string | null;
}
export function getTechniques(projectId: number) {
  return api.get<any, { data: Technique[]; code: number }>('/techniques', { params: { projectId } })
}
export function getArtifacts(projectId: number) {
  return api.get<any, { data: Artifact[]; code: number }>('/artifacts', { params: { projectId } })
}